File: app/src/main/java/com/christianfoulcard/android/androidsensorengine/MainActivity.java
package com.christianfoulcard.android.androidsensorengine;

import android.app.Activity;
import android.app.ActivityOptions;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.WindowManager;
import android.widget.ImageView;

import com.christianfoulcard.android.androidsensorengine.Menu_Items.About;
import com.christianfoulcard.android.androidsensorengine.Menu_Items.Credits;
import com.christianfoulcard.android.androidsensorengine.Menu_Items.Premium;
import com.christianfoulcard.android.androidsensorengine.Preferences.SettingsActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.AccelerometerActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.BatteryActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.HumidityActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.LightSensorActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.PressureActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.RamActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.SoundSensorActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.TemperatureActivity;
import com.christianfoulcard.android.androidsensorengine.Sensors.WalkActivity;
import com.google.firebase.analytics.FirebaseAnalytics;

////////////////////////////////////////////////////////////////////////////////////////////////////

public class MainActivity extends AppCompatActivity {

    //TODO: Show user a list of sensors their device can use
    //TODO: Add elevation/sea level sensor?
    //TODO: Go through each activity's lifecycle
    //TODO: Fix animations
    //TODO: Fix/String in other languages
    //TODO Custom Sounds?
    //TODO: Notification screen cleanup
    //TODO: Fix Pressure Notifications

    ////////////////////////////////////////////////////////////////////////////////////////////////

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        // Make sure this theme is before calling super.onCreate
        setTheme(R.style.AppTheme);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.sensor_selection);

        // This will make the Status Bar completely transparent
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS);

        // Obtain the FirebaseAnalytics instance and Initiate Firebase Analytics
        FirebaseAnalytics mFirebaseAnalytics = FirebaseAnalytics.getInstance(this);
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // Intents to start the sensor activities

    public void soundIconIntent(View view) {
        Intent soundIntent = new Intent(this, SoundSensorActivity.class);

        ImageView sharedView = findViewById(R.id.sound_icon);
        String transitionName = getString(R.string.sound_anim);

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(soundIntent, transitionActivityOptions.toBundle());
    }

    public void tempIconIntent(View view) {
        Intent tempIntent = new Intent(this, TemperatureActivity.class);

        ImageView sharedView = findViewById(R.id.temp_icon);
        String transitionName = getString(R.string.temp_anim);

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(tempIntent, transitionActivityOptions.toBundle());
    }

    public void lightIconIntent(View view) {
        Intent lightIntent = new Intent(this, LightSensorActivity.class);

        ImageView sharedView = findViewById(R.id.light_icon);
        String transitionName = getString(R.string.light_anim);

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(lightIntent, transitionActivityOptions.toBundle());
    }

    public void ramIconIntent(View view) {
        Intent ramIntent = new Intent(this, RamActivity.class);

        ImageView sharedView = findViewById(R.id.ram_icon);
        String transitionName = getString(R.string.ram_anim);

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(ramIntent, transitionActivityOptions.toBundle());
    }

    public void batteryIconIntent(View view) {
        Intent batteryIntent = new Intent(this, BatteryActivity.class);

        ImageView sharedView = findViewById(R.id.battery_icon);
        String transitionName = getString(R.string.battery_anim);

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(batteryIntent, transitionActivityOptions.toBundle());
    }

    public void speedIconIntent(View view) {
        Intent speedIntent = new Intent(this, AccelerometerActivity.class);

        ImageView sharedView = findViewById(R.id.speed_icon);
        String transitionName = getString(R.string.speed_anim);

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(speedIntent, transitionActivityOptions.toBundle());
    }

    public void humidityIconIntent(View view) {
        Intent humidityIntent = new Intent(this, HumidityActivity.class);

        ImageView sharedView = findViewById(R.id.humidity_icon);
        String transitionName = "";

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(humidityIntent, transitionActivityOptions.toBundle());
    }

    public void pressureIconIntent(View view) {
        Intent pressureIntent = new Intent(this, PressureActivity.class);

        ImageView sharedView = findViewById(R.id.pressure_icon);
        String transitionName = "";

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(pressureIntent, transitionActivityOptions.toBundle());
    }

    public void walkIconIntent(View view) {
        Intent walkIntent = new Intent(this, WalkActivity.class);

        ImageView sharedView = findViewById(R.id.walk_icon);
        String transitionName = "";

        ActivityOptions transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, sharedView, transitionName);
        this.startActivity(walkIntent, transitionActivityOptions.toBundle());
    }
}

File: app/src/main/java/com/christianfoulcard/android/androidsensorengine/MainActivity.kt
package com.christianfoulcard.android.androidsensorengine

import android.app.ActivityOptions
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.WindowManager
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import com.christianfoulcard.android.androidsensorengine.BuildConfig.DEBUG
import com.christianfoulcard.android.androidsensorengine.Sensors.*
import com.christianfoulcard.android.androidsensorengine.databinding.SensorSelectionBinding
import com.christianfoulcard.android.androidsensorengine.databinding.SoundSensorBinding
import com.google.firebase.analytics.FirebaseAnalytics
import timber.log.Timber
import timber.log.Timber.DebugTree

//TODO: Show user a list of sensors their device can use
//TODO: Add elevation/sea level sensor?
//TODO: Fix animations
//TODO: Update ram activity parsing



////////////////////////////////////////////////////////////////////////////////////////////////////
class MainActivity : AppCompatActivity() {

    //View Binding to call the layout's views
    private lateinit var binding: SensorSelectionBinding

    // Initiate Firebase Analytics
    private var mFirebaseAnalytics: FirebaseAnalytics? = null

    override fun onCreate(savedInstanceState: Bundle?) {

        // Make sure this theme is before calling super.onCreate
        setTheme(R.style.AppTheme)
        super.onCreate(savedInstanceState)
        binding = SensorSelectionBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        // This will make the Status Bar completely transparent
        window.setFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS,
                WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS)

//        if (BuildConfig.DEBUG) {
//            Timber.plant(DebugTree())
//        }

        // Obtain the FirebaseAnalytics instance.
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(this)
    }

    override fun onStart() {
        super.onStart()
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////
    // Intents to start the sensor activities

    fun soundIconIntent(view: View?) {
        val soundIntent = Intent(this, SoundSensorActivity::class.java)
        val transitionName = getString(R.string.sound_anim)
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.soundIcon, transitionName)
        this.startActivity(soundIntent, transitionActivityOptions.toBundle())
    }

    fun tempIconIntent(view: View?) {
        val tempIntent = Intent(this, TemperatureActivity::class.java)
        val transitionName = getString(R.string.temp_anim)
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.tempIcon, transitionName)
        this.startActivity(tempIntent, transitionActivityOptions.toBundle())
    }

    fun lightIconIntent(view: View?) {
        val lightIntent = Intent(this, LightSensorActivity::class.java)
        val transitionName = getString(R.string.light_anim)
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.lightIcon, transitionName)
        this.startActivity(lightIntent, transitionActivityOptions.toBundle())
    }

    fun ramIconIntent(view: View?) {
        val ramIntent = Intent(this, RamActivity::class.java)
        val transitionName = getString(R.string.ram_anim)
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.ramIcon, transitionName)
        this.startActivity(ramIntent, transitionActivityOptions.toBundle())
    }

    fun batteryIconIntent(view: View?) {
        val batteryIntent = Intent(this, BatteryActivity::class.java)
        val transitionName = getString(R.string.battery_anim)
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.batteryIcon, transitionName)
        this.startActivity(batteryIntent, transitionActivityOptions.toBundle())
    }

    fun speedIconIntent(view: View?) {
        val speedIntent = Intent(this, AccelerometerActivity::class.java)
        val transitionName = getString(R.string.speed_anim)
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.speedIcon, transitionName)
        this.startActivity(speedIntent, transitionActivityOptions.toBundle())
    }

    infix fun humidityIconIntent(view: View?) {
        val humidityIntent = Intent(this, HumidityActivity::class.java)
        val transitionName = ""
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.humidityIcon, transitionName)
        this.startActivity(humidityIntent, transitionActivityOptions.toBundle())
    }

    fun pressureIconIntent(view: View?) {
        val pressureIntent = Intent(this, PressureActivity::class.java)
        val transitionName = ""
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.pressureIcon, transitionName)
        this.startActivity(pressureIntent, transitionActivityOptions.toBundle())
    }

    fun walkIconIntent(view: View?) {
        val walkIntent = Intent(this, WalkActivity::class.java)
        val transitionName = ""
        val transitionActivityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, binding.walkIcon, transitionName)
        this.startActivity(walkIntent, transitionActivityOptions.toBundle())
    }
}


